### PR TITLE
Issue 1093 olli

### LINKED ISSUE #1093: Created glyph uses different case than the entered glyph name
I wanted to create a new glyph called /h.ss01. I double clicked on the new glyph and fontra created a new glyph called /H.ss01. So, it changed it to uppercase. 
Screenshot 2024-02-16 at 09 57 29
Screenshot 2024-02-16 at 09 58 01

### PR DESCRIPTION
**Pull request by ollimeier**
This fixes #1093.

Screenshot 2024-02-19 at 16 42 42

### CODE REVIEW

File: src/fontra/client/core/glyph-lines.js [hunk 1]
```diff
@@ -54,14 +54,23 @@ async function glyphNamesFromText(text, characterMap, glyphMap) {
...
             // a glyph name associated with that character
-            let properBaseGlyphName = characterMap[baseCharCode];
-            if (!properBaseGlyphName) {
-              properBaseGlyphName = await getSuggestedGlyphName(baseCharCode);
-            }
-            if (properBaseGlyphName) {
-              glyphName = properBaseGlyphName + extension;
-              if (!extension) {
-                char = charString;
+            if (baseGlyphName.search(/[^a-zA-Z]+/) === -1) {
```
Comment (justvanrossum) [inline src/fontra/client/core/glyph-lines.js:57]: I don't think this is correct. You are searching for any non-A-Z and non-a-z chars, but that is not the right criterium.

I think the preceding `if (baseGlyphName === charString)` condition should be changed to something like:

```js
    if (baseGlyphName === charString && !isPlainLatinLetter(baseGlyphName) ) {
```

Where `isPlainLatinLetter(glyphName)` checks whether the name is exactly one character, matching `[A-Za-z]`. For example (untested):

```js
function isPlainLatinLetter(glyphName) {
  return glyphName.match(/^[A-Za-z]$/)
}
```

Then the rest of the code can remain unchanged.

**Review by justvanrossum — COMMENTED** [review]
The glyphnames.py change is good, I left a suggestion how to approach .notdef.

File: src/fontra/client/core/glyph-lines.js (hunk 1 above)
Comment (ollimeier) [reply src/fontra/client/core/glyph-lines.js:57]: Where should we store the new function `isPlainLatinLetter`?

Comment (ollimeier) [reply src/fontra/client/core/glyph-lines.js:57]: Within utils?

Comment (justvanrossum) [reply src/fontra/client/core/glyph-lines.js:57]: Or at the end of the module.

Comment (ollimeier) [reply src/fontra/client/core/glyph-lines.js]: Please review: https://github.com/googlefonts/fontra/pull/1100/commits/28feb5d3fb22425aaec603ea3a398236b7500e42

**Pull request merged**